veejay-current/veejay-server/veejay/vj-event.c

```diff
@@ -4153,6 +4180,68 @@ void vj_event_sample_set_dup(void *ptr, const char format[], va_list ap)
     }
 }
 
+void vj_event_mixing_sample_set_speed(void *ptr, const char format[], va_list ap)
+{
+    int args[1];
+    veejay_t *v = (veejay_t*) ptr;
+    int sample_id = -1;
+    char *s = NULL;
+    P_A(args, s, format, ap);
+
+    if(SAMPLE_PLAYING(v))
+    {
+        int entry = sample_get_selected_entry(v->uc->sample_id);
+        sample_id = sample_get_chain_channel(v->uc->sample_id,entry);
```

Comment:
You need to take a look at the mixing source type; If the sample is mixing with stream 3, this event will actually update the speed of sample 3.